app/src/main/java/org/listenbrainz/android/util/ListenSessionListener.kt

```diff
@@ -26,28 +29,45 @@ class ListenSessionListener(
         clearSessions()
 
         val token = appPreferences.lbAccessToken
-        if (!token.isNullOrEmpty()) {
+        val username = appPreferences.username
+        if (token.isNullOrEmpty()) {
             d("ListenBrainz User token has not been set!")
             return
         }
 
-//        service.getServicesLinkedToAccount(token, appPreferences.username!!)?.enqueue(object : retrofit2.Callback<ListenBrainzExternalServices?> {
-//            override fun onResponse(call: Call<ListenBrainzExternalServices?>, response: Response<ListenBrainzExternalServices?>) {
-//                println("Services found: " + response.body())
-//                registerControllers(controllers, services = response.body())
-//            }
-//            override fun onFailure(call: Call<ListenBrainzExternalServices?>, t: Throwable) {
-//                println("Services found not found")
-//                registerControllers(controllers)
-//            }
-//        })
+        if (username != null) {
```

Comment:
This method gets called a lot so I would suggest making a sharedPreference and making an API call when the app starts to update it. I think we could improve this with FCM later (idkfs) but for now I think this would be the way.

Reply:
We don't have an API which we could automatically call to get the username. The user has to login to MB first, only then we can get the username. Once that is done, we do save it in the shared preferences and fetch it from there only. (Have a look at line 32)
FCM won't help us with this.

Reply:
Shouldn't it just be `if (appPreferences.username)` as its a getter property. This means username variable will always be null even if user logs on since ListensService never dies.

Reply:
Preferences can be nullable by default @07jasjeet

Reply:
No no, I mean here username variable won't change whatever the value be. This is because username variable will be runtime variable for ListenService and getters only work access them directly.

Reply:
Right, I get what you mean now. The fact that username variable won't be mutable in this condition, so even after logging in, the service won't start until and the unless the user kills the app and restarts.

Reply:
Ugh. Good catch. I don't want a viewmodel here just yet. But we might have to do it now.

Reply:
@07jasjeet Even with `if (appPreferences.username)` this won't work. We need an observable state here.

Reply:
Wait wait, its fine (I didn't unwrap the code). But still the main problem is the first argument. Come on IRC